src/main/java/me/xemor/superheroes/skills/skilldata/DamageModifierData.java

```diff
@@ -0,0 +1,102 @@
+package me.xemor.superheroes.skills.skilldata;
+
+import me.xemor.superheroes.configurationdata.comparison.SetData;
+import me.xemor.superheroes.org.jetbrains.annotations.NotNull;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.entity.EntityDamageEvent;
+
+public class DamageModifierData extends SkillData {
+
+    private final SetData<EntityType> entitywhitelist;
+
+    private final SetData<EntityType> entityblacklist;
+
+    private final SetData<EntityDamageEvent.DamageCause> causeblacklist;
+
+    private final SetData<EntityDamageEvent.DamageCause> causewhitelist;
+
+
+    private final double expectedMaximumDamage;
+
+    private final double damageCap;
+
+    private final double damageMin;
+    private final int priority;
+    private final boolean incoming;
+    private final boolean outgoing;
+    private final boolean eased;
+
+    public DamageModifierData(int skill,
+                              @NotNull ConfigurationSection configurationSection) {
+        super(skill, configurationSection);
+
+        expectedMaximumDamage = configurationSection.getDouble("expectedMaxDamage", 30);
+        damageCap = configurationSection.getDouble("damageMax", 15);
+        damageMin = configurationSection.getDouble("damageMin", 0);
+        causeblacklist =
+            new SetData<>(EntityDamageEvent.DamageCause.class, "causeblacklist",
+                configurationSection);
+        causewhitelist =
```

Comment:
tbh I'd be tempted to just have "causes" and "entities" then a toggle to swap between blacklist and whitelist, I think this is how I normally do it

Reply:
then have blacklist on true by default (assume if they don't specify, it allows all)

Reply:
whitelist: false is on by default